code/components/extra-natives-rdr3/src/ShapeTestNatives.cpp

```diff
@@ -0,0 +1,98 @@
+#include <StdInc.h>
+#include <ScriptEngine.h>
+
+#include <Hooking.h>
+#include <scrEngine.h>
+#include <EntitySystem.h>
+#include <ScriptSerialization.h>
+
+struct ShapeTestResult
+{
+	uint32_t shapeTestHandle;
+	bool hit;
+	scrVector endCoords;
+	scrVector surfaceNormal;
+	uint64_t entityHit;
+};
+
+class ResultData
+{
+public:
+	char pad_0000[108]; // 0x0000
+	uint8_t materialIndex; // 0x006C
+}; // Size: 0x006D
+
+class CScriptShapeTestResult
+{
+public:
+	char pad_0000[36]; // 0x0000
+	uint32_t state; // 0x0024
+	char pad_0028[8]; // 0x0028
+	class ResultData* data; // 0x0030
+}; // Size: 0x0038
+
+template<std::size_t Index, typename ReturnType, typename... Args>
+inline ReturnType call_virtual(void* instance, Args... args)
+{
+	using Fn = ReturnType(__thiscall*)(void*, Args...);
+
+	auto function = (*reinterpret_cast<Fn**>(instance))[Index];
+	return function(instance, args...);
+}
+
+static hook::cdecl_stub<CScriptShapeTestResult*(uint64_t)> g_getCScriptShapeTestResult([]()
+{
+	return hook::get_call(hook::get_pattern("E9 ? ? ? ? 90 F3 0F 8B 05 1A 2D BB"));
+});
+
+// This information was previously returned into the native, but this is not the case for RDR3
+// we need to grab it from the trace result.
+uint8_t GetMateralIndexFromTraceHandle(uint64_t rayHandle)
+{
+	CScriptShapeTestResult* shapeResult = g_getCScriptShapeTestResult(rayHandle);
+
+	if (!shapeResult || !shapeResult->data)
+	{
+		return 0;
+	}
+
+	return shapeResult->data->materialIndex;
+}
+
+static HookFunction hookFunction([]()
+{
+	static void** phMaterialMgr = hook::get_address<void**>(hook::get_pattern("48 89 05 ? ? ? ? 48 83 C4 28 C3 90 78 48"), 3, 7);
```

Comment:
This pattern is invalid for 1311 and 1355 game builds. You included a part of the referring address. I replaced your pattern with this:
```cpp
static void** phMaterialMgr = hook::get_address<void**>(hook::get_pattern("48 8B CB 0F 28 CA F3 0F 59", 22));
```